packages/react-reconciler/src/ReactStrictModeWarnings.js

```diff
@@ -289,6 +296,70 @@ if (__DEV__) {
       });
     }
   };
+
+  ReactStrictModeWarnings.recordLegacyContextWarning = (
+    fiber: Fiber,
+    instance: any,
+  ) => {
+    // Dedup strategy: Warn once per component.
+    if (didWarnAboutLegacyContext.has(fiber.type)) {
+      return;
+    }
+
+    const strictRoot = findStrictRoot(fiber);
+    if (strictRoot === null) {
+      warning(
+        false,
+        'Expected to find a StrictMode component in a strict mode tree. ' +
+          'This error is likely caused by a bug in React. Please file an issue.',
+      );
+      return;
+    }
+
+    let warningsForRoot = pendingLegacyContextWarning.get(strictRoot);
+    if (warningsForRoot === undefined) {
+      warningsForRoot = [];
+      pendingLegacyContextWarning.set(strictRoot, warningsForRoot);
+    }
+
+    if (
+      typeof instance.getChildContext === 'function' ||
+      fiber.type.contextTypes != null ||
+      fiber.type.childContextTypes != null
+    ) {
+      warningsForRoot.push(fiber);
+      pendingLegacyContextWarning.set(strictRoot, warningsForRoot);
```

Comment:
nit: We don't actually need to re-set this in the `Map` do we?

Reply:
Is it because when we do `let warningsForRoot = pendingLegacyContextWarning.get(strictRoot);`

and `warningsForRoot.push` is mutating the content in `pendingLegacyContextWarning`?

Reply:
I thought it was creating a new copy. I'll delete this line.

Reply:
Right. You're updating the reference that's already stored in the `Map`.